extensions/blocks/calendly/edit.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * External Dependencies
+ */
+import 'url-polyfill';
+import { isEqual } from 'lodash';
+import queryString from 'query-string';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import {
+	Button,
+	ExternalLink,
+	Notice,
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import icon from './icon';
+import attributeDetails, { getValidatedAttributes } from './attributes';
+import SubmitButton from '../../shared/submit-button';
+import { getAttributesFromEmbedCode } from './utils';
+import blockStylesPreviewAndSelector from './blockStylesPreviewAndSelector';
+
+export default function CalendlyEdit( { attributes, className, setAttributes } ) {
+	const validatedAttributes = getValidatedAttributes( attributeDetails, attributes );
+
+	if ( ! isEqual( validatedAttributes, attributes ) ) {
+		setAttributes( validatedAttributes );
+	}
+
+	const {
+		backgroundColor,
+		submitButtonText,
+		hideEventTypeDetails,
+		primaryColor,
+		textColor,
+		style,
+		url,
+	} = validatedAttributes;
+	const [ embedCode, setEmbedCode ] = useState();
+	const [ notice, setNotice ] = useState();
+
+	const setErrorNotice = () =>
+		setNotice(
+			<>
+				{ __(
+					"Your calendar couldn't be embedded. Please double check your URL or code.",
+					'jetpack'
+				) }
+			</>
+		);
+
+	const parseEmbedCode = event => {
+		if ( ! event ) {
+			setErrorNotice();
+			return;
+		}
+
+		event.preventDefault();
+
+		const newAttributes = getAttributesFromEmbedCode( embedCode );
+		if ( ! newAttributes ) {
+			setErrorNotice();
+			return;
+		}
+
+		const newValidatedAttributes = getValidatedAttributes( attributeDetails, newAttributes );
+
+		setAttributes( newValidatedAttributes );
+	};
+
+	const embedCodeForm = (
+		<form onSubmit={ parseEmbedCode }>
+			<input
+				type="text"
+				id="embedCode"
+				onChange={ event => setEmbedCode( event.target.value ) }
+				placeholder={ __( 'Calendly web address or embed code…' ) }
```

Comment:
I think we'll need a `jetpack` textdomain here.

```suggestion
				placeholder={ __( 'Calendly web address or embed code…', 'jetpack' ) }
```